### DISCUSSION TITLE
Build without GMP and MPFR

### DISCUSSION THREAD

**Discussion by tuket** [libigl/libigl#2428] [Q&A]
I would like to compile IGL without GMP, and without MPFR.
These are dependencies of CGAL but they are supposed to be optional.
I have seen that, internally, CGAL checks for the CGAL_DISABLE_GMP environment variable. I tried setting it in the CMakeLists.txt but it didn't work.
How can I disable those optional libraries?

**Comment by alecjacobson** [comment]
They are not optional for the parts of cgal that the libigl cgal module
uses. You can disable the libigl cgal module instead with
LIBIGL_COPYLEFT_CGAL=OFF in cmake

**Reply by tuket** [reply]
Hi, thanks for your answer!
I need CGAL because I need to do boolean operations.
I thought GMP wasn't required at all by CGAL, at least for these boolean operations.
The reason I wanted to disable GMP is because I wanted to compile for WebAssembly (using Emscripten), so GMP would seem like hard to get compiled on the web (specially since the project doesn't even use CMake).
